libfreerdp/core/nla.c

```diff
@@ -1085,6 +1085,53 @@ BOOL nla_decrypt_public_key_hash(rdpNla* nla)
 	return status;
 }
 
+static BOOL set_creds_octetstring_to_settings(WinPrAsn1Decoder* dec, WinPrAsn1_tagId tagId,
+                                              BOOL optional, size_t settingId,
+                                              rdpSettings* settings)
+{
+	if (optional)
+	{
+		WinPrAsn1Decoder ctxtDec;
+		WinPrAsn1_tagId itemTag;
+		if (!WinPrAsn1DecPeekContextualTag(dec, &itemTag, &ctxtDec) || (itemTag != tagId))
+			return TRUE;
+	}
+
+	BOOL error = FALSE;
+	WinPrAsn1_OctetString value;
+	if (!WinPrAsn1DecReadContextualOctetString(dec, tagId, &error, &value, FALSE) && error)
+		return FALSE;
+
+	return freerdp_settings_set_string_from_utf16N(settings, settingId, (const WCHAR*)value.data,
```

Comment:
might be handy to redefine `WinPrAsn1_OctetString` so that you can directly access the data as `char*` or `WCHAR*`

Reply:
Good idea, but that's probably for another PR because we have quite a lot of OctetStrings with ASN1.